### ISSUE TITLE
Listing all cells and resolutions

### ISSUE THREAD

**Issue by aaime** [riskaware-ltd/open-eaggr#2]
Is there any way to list all cells for the base resolution level (the most coarse one, the base cells)?
Is there a way to get a list of of available resolutions?

**Comment by ian-bush-riskaware** [comment]
Hi Andrea - thanks for your interest in OpenEAGGR. The base cell resolution in OpenEAGGR is at the level of the icosahedron that models the globe - i.e. each face is a base cell. The first two digits of any cell id indicates the face of the icosahedron that the cell is located on, and the base cell is just the face id. So for the icosahedron there are 20 base cells.
The number of resolutions is limited only by the maximum length of the cell id string (around 40 characters). The area of each cell is dependent on the aperture of the chosen model, so for ISEA4T the area of each cell is 1/4 of the parent cell; for ISEA3H it is 1/3. The base cell area is 1/20 of the area of the Earth (as the icosahedron has 20 faces) and each resolution will be correspondingly smaller.
In OpenEAGGR you don't really worry about specifying the resolution but each point that you convert to a DGGS cell id has an associated 'accuracy area' in metres squared which determines which resolution cell is created. The idea is the size of your cell is related to the accuracy of your data.
Please get it touch again if you have any further questions.

**Comment by aaime** [comment]
Thanks for the follow up. So for the ISEA4T model the base cells are named "00" to "19" right?

What about the ISEA3H model instead? The names seem strange, for example "07000,0" seems to be a base cell, and "[07010,0, 0701-1,0, 07010,1, 07011,0, 07010,-1, 0701-1,1, 0701-1,-1]" its children (as returned by the getCellChildren method). Is this nomenclature documented anywhere?

**Comment by ian-bush-riskaware** [comment]
Yes you are correct. the base cells for ISEA4T are "00" to "19". For ISEA3H the indexing uses an offset system rather than a hierarchical system so the first two digits are again the face of the icosahedron 0 to 19, the next two digits are the resolution level - "00" for the base level and then a coordinate of the cell on the face which will be "0,0" for the base cells. The child cells of the base cell have resolution level "01" and offsets around the central hexagon.

The coordinate systems are described in the [Literature Review and Protoype Evaluation](https://github.com/riskaware-ltd/open-eaggr/blob/master/Documents/Literature%20Review%20%26%20Prototype%20Evaluation.pdf) document in section 8.
